Threads/UARTCommsController.cpp

```diff
@@ -6,330 +6,337 @@
-#define isEndOfCommand(x) 	(x == 10 || x == 13)		// LF (10) or CR (13)
-#define isCheckSum(x)		(x == ';')
-#define isService(x)		(x == '(' || x == ')')
-#define isValid(x)			(isEndOfCommand(x) || isprint(x))
 
-/* cunning way to convert character into a digit. This is possible due to ASCII table digit mapping. */
-#define toDigit(x)			(x - 48)
+#define RESPONSE_MARKER '<'
+#define COMMAND_MARKER '>'
+#define isCommandMarker(x) (x == COMMAND_MARKER)
+#define isEndOfCommand(x) (x == 10 || x == 13) // LF (10) or CR (13)
+#define isCheckSum(x) (x == ';')
+#define isService(x) (x == '(' || x == ')')
+#define isValid(x) (isEndOfCommand(x) || isprint(x))
 
-UARTCommsController* UARTCommsController::s_pUARTController1 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController2 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController3 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController4 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController5 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController6 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController7 = NULL;
-UARTCommsController* UARTCommsController::s_pUARTController8 = NULL;
+/* cunning way to convert character into a digit. This is possible due to ASCII
+ * table digit mapping. */
+#define toDigit(x) (x - 48)
+
+UARTCommsController *UARTCommsController::s_pUARTController1 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController2 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController3 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController4 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController5 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController6 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController7 = NULL;
+UARTCommsController *UARTCommsController::s_pUARTController8 = NULL;
 
 /* TxBuffer is large to accommodate possible large transmissions
  * RxBuffer is smaller since it only needs to handle incoming commands
  * @115200 baud it'll take 1/115200 * (1 + 8 + 1) * 64 = 5.5 ms
  * to overflow. Since this controller runs faster than that, it should
  * never happen. */
-UARTCommsController::UARTCommsController() :
-m_pHUART(NULL),
-m_TxBuffer(200),
-m_RxBuffer(64),
-m_ControllerCount(0)
-{
-	m_pControllers[m_ControllerCount] = NULL;
-	m_IncomingRxChar = '\0';
-	m_NewCommand[49] = '\0';
-	m_RxCRCString[2] = '\0';
-	Reset();
+UARTCommsController::UARTCommsController()
+    : m_pHUART(NULL), m_TxBuffer(200), m_RxBuffer(64), m_ControllerCount(0),
+      m_bTestVariable(false) {
+  m_pControllers[m_ControllerCount] = NULL;
+  m_IncomingRxChar = '\0';
+  m_NewCommand[49] = '\0';
+  m_RxCRCString[2] = '\0';
+  Reset();
 }
 
-void UARTCommsController::Initialise(UART_HandleTypeDef * pHUART, bool debugChannel){
-	m_pHUART = pHUART;
-	m_bDebugChannel = debugChannel;
+void UARTCommsController::Initialise(UART_HandleTypeDef *pHUART,
```

Comment:
@Mustafa, please, check that this class initialisation fixes all uninitialised variables.

Reply:
OK. Undrestood. Well done.